### PR TITLE
mon/OSDMonitor: clear stale in-memory prune state

### PR DESCRIPTION
**Pull request by jecluis**
Fixes: http://tracker.ceph.com/issues/24612

Signed-off-by: Joao Eduardo Luis <joao@suse.de>

### CODE REVIEW

**Comment by jecluis** [comment]
Please disregard for the time being. Will push a new patch soon. This one is a rather silly kludge. The behavior being fixed atm is ensuring we don't alter the in-memory state of the osdmap manifest unless it's coming directly from update_from_paxos().

**Comment by jecluis** [comment]
New patch is up.

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -2019,8 +2021,8 @@ bool OSDMonitor::do_prune(MonitorDBStore::TransactionRef tx)
   ceph_assert(num_pruned > 0);
 
   bufferlist bl;
-  osdmap_manifest.encode(bl);
-  tx->put(get_service_name(), "osdmap_manifest", bl);
+  manifest.encode(bl);
+  tx->put(get_service_name(), "manifest", bl);
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: probably didn't mean to change the key name?

Comment (jecluis) [reply src/mon/OSDMonitor.cc]: oops. that's what you get for a blanket replace. thanks for catching that. :)

**Comment by tchaikov** [comment]
shall we update `_prune_update_trimmed()` as well?

**Comment by jecluis** [comment]
@tchaikov done. Also, I'm currently without access to the sepia lab (for reasons only openvpn client understands, will sort it out soon), and unable to run this through suites. Mind scheduling a thrasher for me?

**Comment by tchaikov** [comment]
@jecluis will do.

**Comment by tchaikov** [comment]
- rados:thrash: http://pulpito.ceph.com/kchai-2018-08-28_12:33:22-rados:thrash-wip-kefu-testing-2018-08-28-0811-distro-basic-smithi/
- rados: http://pulpito.ceph.com/kchai-2018-08-28_04:25:25-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-08-28_04:25:25-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_07:26:03-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_11:04:40-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_13:22:00-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2018-08-29_15:16:19-rados-wip-kefu-testing-2018-08-28-0811-distro-basic-mira/

fixes posted at #23681 and #23803 


@liewegas did Joao address your comment?

**Pull request merged**